### PR TITLE
Update logging and errors

### PR DESCRIPTION
**Pull request by ltseeley**
Numerous improvements to comments, docstrings, log messages, and error handling

### CODE REVIEW

**Comment by peterschwarz** [comment]
Actually meant to request changes there

File: src/engine.rs [hunk 1]
```diff
@@ -142,20 +146,23 @@ fn handle_update(
...
             node.on_peer_message(parsed_message, state)?
         }
-        Ok(Update::Shutdown) => return Ok(false),
+        Ok(Update::Shutdown) => {
+            info!("Received shutdown; stopping PBFT");
+            return Ok(false);
+        }
         Ok(Update::PeerConnected(_)) | Ok(Update::PeerDisconnected(_)) => {
-            debug!("Received PeerConnected/PeerDisconnected message");
+            info!("Received PeerConnected/PeerDisconnected message");
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply src/engine.rs]: Fixed

File: src/node.rs [hunk 2]
```diff
@@ -105,7 +104,7 @@ impl PbftNode {
             PbftMessageType::Commit => self.handle_commit(msg, state)?,
             PbftMessageType::ViewChange => self.handle_view_change(&msg, state)?,
             PbftMessageType::NewView => self.handle_new_view(&msg, state)?,
-            _ => warn!("Message type not implemented"),
+            _ => warn!("Received message with unknown type"),
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply src/node.rs]: Fixed

**Comment by chenette** [comment]
The comments, docstrings, and log messages look good.

File: src/engine.rs [hunk 3]
```diff
@@ -94,7 +94,7 @@ impl Engine for PbftEngine {
                 // ViewChange if necessary
                 if node.check_faulty_primary_timeout_expired(state) {
                     warn!("Faulty primary timeout expired; proposing view change");
-                    handle_pbft_result(node.start_view_change(state, state.view + 1));
+                    log_any_error(node.start_view_change(state, state.view + 1));
```
Comment (vaporos) [inline src/engine.rs]: These log_any_error() calls seem a bit awkward. In any case, this might be a good use for a macro.

**Comment by ltseeley** [comment]
Removed the 2 commits that renamed `InternalError` to `GeneralError` and removed the `SerializationError` type. Added 5 commits to properly categorize and display the different error types.

File: src/engine.rs (hunk 3 above)
Comment (ltseeley) [reply src/engine.rs]: Could you explain what makes this a good use for a macro? Or do you have other suggestions for how to make this less awkward? The only thing I can think of at the moment would be to just call `.unwrap_or_else(|err| error!("{}", err)` in these situations.

**Pull request merged**